apps/aecore/lib/aecore/chain/chain_state.ex

```diff
@@ -83,32 +72,22 @@ defmodule Aecore.Chain.ChainState do
     valid_txs_list
   end
 
-  @spec validate_tx(SignedTx.t(), chainstate(), integer()) :: {boolean(), chainstate()}
-  defp validate_tx(tx, chainstate, block_height) do
+  @spec validate_tx(SignedTx.t(), chainstate()) :: {boolean(), chainstate()}
+  defp validate_tx(tx, chainstate) do
     try do
-      {true, apply_transaction_on_state!(tx, chainstate, block_height)}
+      {true, apply_transaction_on_state!(tx, chainstate)}
     catch
       {:error, _} -> {false, chainstate}
     end
   end
 
   @spec calculate_total_tokens(chainstate()) :: non_neg_integer()
   def calculate_total_tokens(%{accounts: accounts}) do
-    Enum.reduce(accounts, 0, fn {_account, state}, acc ->
-      acc + state.balance
+    AccountStateTree.reduce(accounts, 0, fn {k, _v}, acc ->
```

Comment:
Would be better if you rename {k, _v} to {key, _val} for more clarity

Reply:
best would be semantic correct names